macos/Sources/Features/Terminal/BaseTerminalController.swift

```diff
@@ -548,6 +553,11 @@ class BaseTerminalController: NSWindowController,
         window.performClose(sender)
     }
 
+    @IBAction func toggleMaximize(_ sender: Any) {
+        guard let window = window else { return }
```

Comment:
There needs to be a guard here that the object is within the `surfaceTree` otherwise toggle maximize will maximize all open terminal windows. 😄 

There should be at least one other action that has this check, search for `contains`.